safe-global/safe-wallet-web
# feat: recipient address security module

## What it solves

Resolves #2014

## How this PR fixes it

This adds the `RecipientAddressModule`, as well as associated hook and demo implementation in the transaction modal. It covers normal/multiSend transfers of native, ERC-20 and ERC-721 assets.

The implementation checks and warns of the following:

1. Whether the address is either an owner or present in the address book.
2. Whether the address is "used", e.g. it is an EoA with native balance.
3. Whether the address is a Safe deployed otherwise on mainnet (the user is sending funds to the wrong chain).

## How to test it

Create a transaction with the recipients matching either of the three above criteria and observe the warning(s).

## Screenshots

Please note that this is a demo implementation.

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/eb3e7760-267a-4036-9a9d-ad51c830175b)

## Checklist
